src/Component.php

```diff
@@ -150,7 +150,7 @@ protected function methodForm($method, &$form_state) {
     $form_elements_callback = $method->controller->payment_configuration_form_elements_callback;
     if (function_exists($form_elements_callback) == TRUE) {
       $form_state['payment'] = $payment;
-      $element += $form_elements_callback($element, $form_state);
+      $element = $form_elements_callback($element, $form_state);
```

Comment:
Have you checked there aren’t any payment_configuration_form_elements_callbacks that expect to extend $element and would loose part of their $element settings with that change?

Reply:
Those are:

- [payment_forms_pament_form()](https://github.com/moreonion/payment_forms/blob/7dd17e0c3b9aefda97dee11bb8b8ef5898c24443/payment_forms.module#L12) used by all our payment methods (braintree_payment, direct_mind_payment, gocardless_payment, manual_direct_debit, paymill_payment, payone_payment, sagepay_payment, stripe_payment)
- [paypal_payment_ec_payment_method_configuration_form_elements()](https://github.com/moreonion/paypal_payment/blob/a96edbdfb04cb5141b9aface3eb8a0ed5b67c1f6/paypal_payment_ec/paypal_payment_ec.module#L92) and [paypal_payment_pps_payment_method_configuration_form_elements()](https://github.com/moreonion/paypal_payment/blob/a96edbdfb04cb5141b9aface3eb8a0ed5b67c1f6/paypal_payment_pps/paypal_payment_pps.module#L112)

All of them don’t declare the reference in their function signature, which means they can’t edit the `$form`/`$element` by reference.

Reply:
Ah, different change. You were right. All of them create a new form element, that they then return, ignoring everything that’s already in there. So we’ll need another fix for this …

Reply:
… and again wrong on my part. `payment_forms_payment_form()` and most of our payment methods return the passed data as well. Only paypal_payment_* erases everything. That’s why it didn’t show up in my tests. The new code can now handle both cases without needing to change any of the payment methods.